sdk/test/metrics/aggregation_test.cc

```diff
@@ -0,0 +1,122 @@
+TEST(Aggregation, LongSumAggregation)
+{
+  LongSumAggregation aggr(true);
+  auto data = aggr.Collect();
+  ASSERT_TRUE(nostd::holds_alternative<SumPointData>(data));
+  auto sum_data = nostd::get<SumPointData>(data);
+  ASSERT_TRUE(nostd::holds_alternative<long>(sum_data.value_));
+  EXPECT_EQ(nostd::get<long>(sum_data.value_), 0l);
+  EXPECT_EQ(sum_data.is_monotonic_, true);
+  EXPECT_NO_THROW(aggr.Aggregate(12l, {}));
+  EXPECT_NO_THROW(aggr.Aggregate(0l, {}));
+  sum_data = nostd::get<SumPointData>(aggr.Collect());
+  EXPECT_EQ(nostd::get<long>(sum_data.value_), 12l);
+}
+
+TEST(Aggregation, DoubleSumAggregation)
+{
+  DoubleSumAggregation aggr(true);
+  auto data = aggr.Collect();
+  ASSERT_TRUE(nostd::holds_alternative<SumPointData>(data));
+  auto sum_data = nostd::get<SumPointData>(data);
+  ASSERT_TRUE(nostd::holds_alternative<double>(sum_data.value_));
+  EXPECT_EQ(nostd::get<double>(sum_data.value_), 0);
+  EXPECT_EQ(sum_data.is_monotonic_, true);
+  EXPECT_NO_THROW(aggr.Aggregate(12.0, {}));
+  EXPECT_NO_THROW(aggr.Aggregate(1.0, {}));
+  sum_data = nostd::get<SumPointData>(aggr.Collect());
+  EXPECT_EQ(nostd::get<double>(sum_data.value_), 13.0);
+}
+
+TEST(Aggregation, LongLastValueAggregation)
+{
+  LongLastValueAggregation aggr;
+  auto data = aggr.Collect();
+  ASSERT_TRUE(nostd::holds_alternative<LastValuePointData>(data));
+  auto lastvalue_data = nostd::get<LastValuePointData>(data);
+  ASSERT_TRUE(nostd::holds_alternative<long>(lastvalue_data.value_));
+  EXPECT_EQ(lastvalue_data.is_lastvalue_valid_, false);
+  EXPECT_NO_THROW(aggr.Aggregate(12l, {}));
+  EXPECT_NO_THROW(aggr.Aggregate(1l, {}));
+  lastvalue_data = nostd::get<LastValuePointData>(aggr.Collect());
+  EXPECT_EQ(nostd::get<long>(lastvalue_data.value_), 1.0);
+}
+
+TEST(Aggregation, DoubleLastValueAggregation)
+{
+  DoubleLastValueAggregation aggr;
+  auto data = aggr.Collect();
+  ASSERT_TRUE(nostd::holds_alternative<LastValuePointData>(data));
+  auto lastvalue_data = nostd::get<LastValuePointData>(data);
+  ASSERT_TRUE(nostd::holds_alternative<double>(lastvalue_data.value_));
+  EXPECT_EQ(lastvalue_data.is_lastvalue_valid_, false);
+  EXPECT_NO_THROW(aggr.Aggregate(12.0, {}));
+  EXPECT_NO_THROW(aggr.Aggregate(1.0, {}));
+  lastvalue_data = nostd::get<LastValuePointData>(aggr.Collect());
+  EXPECT_EQ(nostd::get<double>(lastvalue_data.value_), 1.0);
+}
+
+TEST(Aggregation, LongHistogramAggregation)
+{
+  LongHistogramAggregation aggr;
+  auto data = aggr.Collect();
+  ASSERT_TRUE(nostd::holds_alternative<HistogramPointData>(data));
+  auto histogram_data = nostd::get<HistogramPointData>(data);
+  ASSERT_TRUE(nostd::holds_alternative<long>(histogram_data.sum_));
+  ASSERT_TRUE(nostd::holds_alternative<std::vector<long>>(histogram_data.boundaries_));
+  EXPECT_EQ(nostd::get<long>(histogram_data.sum_), 0);
+  EXPECT_EQ(histogram_data.count_, 0);
+  EXPECT_NO_THROW(aggr.Aggregate(12l, {}));   // lies in third bucket (indexed at 0)
```

Comment:
The comments are not correct. also in `DoubleHistogramAggregation`

Reply:
Do you mean 12 does lie in 3rd bucket? The default buckets here are: [-infinity to 0], [0 to 5], [5 to10], [10 to 25] , and so on (refer - https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#histogram-aggregation ). So 12 will lie in the third bucket.

Reply:
No I mean `indexed at 0` part, this should be 2 or I'm getting sth wrong here?

Reply:
Yes, have only kept the bucket number now. Thanks for noticing it.